File: java/02186_minimum_number_of_steps_to_make_two_strings_anagram_II.java
//       Java
// Time complexity - O (N)
// Space complexity - O (1)

// example: s = "abc" ; t = "bcd"
// freq: a = 1, b = 1, c = 1
// freq: a = 1, b = 0, c = 0, d = -1

public class Solution {
    public int minSteps(String s, String t) {
        int[] freq = new int[26];

        for (char c : s.toCharArray()) {
            freq[c - 'a']++;
        }

        for (char c : t.toCharArray()) {
            freq[c - 'a']--;
        }

        int steps = 0;
        for (int count : freq) {
            steps += Math.abs(count);
        }

        return steps;
    }
}

File: python/02186_minimum_number_of_steps_to_make_two_strings_anagram_II.py
#       Python
# Time complexity - O (N)
# Space complexity - O (1)

# example: s = "abc" ; t = "bcd"
# freq: a = 1, b = 1, c = 1
# freq: a = 1, b = 0, c = 0, d = -1

class Solution:
    def minSteps(self, s: str, t: str) -> int:
        freq = [0] * 26

        for char in s:
            freq[ord(char) - ord('a')] += 1

        for char in t:
            freq[ord(char) - ord('a')] -= 1

        return sum(abs(count) for count in freq)
